### PR TITLE
feat(closepayment): CHK-1548 add rrn for xpay with transactionId

### PR DESCRIPTION
**Pull request by infantesimone**
#### List of Changes

- add `rrn` set with `transactionId` in `closePayment` for `xpay` authorizations;

#### Motivation and Context

The goal of this PR is to have a `rrn`  set with `transactionId` in `closePayment` for `xpay` authorizations,

#### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: src/test/java/it/pagopa/transactions/utils/AuthRequestDataUtilsTest.java [hunk 1]
```diff
@@ -30,7 +32,9 @@ void shouldExtractVposInformation() {
                                 .rrn("rrn")
                 )
                 .timestampOperation(OffsetDateTime.now());
-        AuthRequestDataUtils.AuthRequestData data = authRequestDataUtils.from(updateAuthorizationRequest);
+        TransactionId transactionId = new TransactionId(UUID.randomUUID().toString().replace("-", ""));
```
Comment (pietro-tota) [inline src/test/java/it/pagopa/transactions/utils/AuthRequestDataUtilsTest.java:35]: ### 
```suggestion
        TransactionId transactionId = new TransactionId(UUID.randomUUID());
```

File: src/test/java/it/pagopa/transactions/utils/AuthRequestDataUtilsTest.java [hunk 2]
```diff
@@ -52,7 +56,9 @@ void shouldExtractVposInformationWithErrorCode() {
                                 .errorCode(OutcomeVposGatewayDto.ErrorCodeEnum._01)
                 )
                 .timestampOperation(OffsetDateTime.now());
-        AuthRequestDataUtils.AuthRequestData data = authRequestDataUtils.from(updateAuthorizationRequest);
+        TransactionId transactionId = new TransactionId(UUID.randomUUID().toString().replace("-", ""));
```
Comment (pietro-tota) [inline src/test/java/it/pagopa/transactions/utils/AuthRequestDataUtilsTest.java:59]: ### 
```suggestion
        TransactionId transactionId = new TransactionId(UUID.randomUUID());
```

File: src/test/java/it/pagopa/transactions/utils/AuthRequestDataUtilsTest.java [hunk 3]
```diff
@@ -78,8 +84,13 @@ void shouldExtractXpayInformation() {
                                 .authorizationCode("authorizationCode")
                 )
                 .timestampOperation(OffsetDateTime.now());
-        AuthRequestDataUtils.AuthRequestData data = authRequestDataUtils.from(updateAuthorizationRequest);
 
+        TransactionId transactionId = new TransactionId(UUID.randomUUID().toString().replace("-", ""));
```
Comment (pietro-tota) [inline src/test/java/it/pagopa/transactions/utils/AuthRequestDataUtilsTest.java:88]:
```suggestion
        TransactionId transactionId = new TransactionId(UUID.randomUUID());
```

File: src/test/java/it/pagopa/transactions/utils/AuthRequestDataUtilsTest.java [hunk 4]
```diff
@@ -99,8 +110,12 @@ void shouldExtractXpayInformationWithErrorCode() {
                                 .errorCode(OutcomeXpayGatewayDto.ErrorCodeEnum.NUMBER_1)
                 )
                 .timestampOperation(OffsetDateTime.now());
-        AuthRequestDataUtils.AuthRequestData data = authRequestDataUtils.from(updateAuthorizationRequest);
+        TransactionId transactionId = new TransactionId(UUID.randomUUID().toString().replace("-", ""));
```
Comment (pietro-tota) [inline src/test/java/it/pagopa/transactions/utils/AuthRequestDataUtilsTest.java:113]:
```suggestion
        TransactionId transactionId = new TransactionId(UUID.randomUUID());
```

File: src/test/java/it/pagopa/transactions/utils/AuthRequestDataUtilsTest.java [hunk 5]
```diff
@@ -78,8 +84,13 @@ void shouldExtractXpayInformation() {
                                 .authorizationCode("authorizationCode")
                 )
                 .timestampOperation(OffsetDateTime.now());
-        AuthRequestDataUtils.AuthRequestData data = authRequestDataUtils.from(updateAuthorizationRequest);
 
+        TransactionId transactionId = new TransactionId(UUID.randomUUID().toString().replace("-", ""));
+        AuthRequestDataUtils.AuthRequestData data = authRequestDataUtils
+                .from(updateAuthorizationRequest, transactionId);
+
+        assertEquals(data.rrn(), ((OutcomeVposGatewayDto) updateAuthorizationRequest.getOutcomeGateway()).getRrn());
```
Comment (pietro-tota) [inline src/test/java/it/pagopa/transactions/utils/AuthRequestDataUtilsTest.java:92]:
```suggestion
```

File: src/test/java/it/pagopa/transactions/utils/AuthRequestDataUtilsTest.java [hunk 6]
```diff
@@ -99,8 +110,12 @@ void shouldExtractXpayInformationWithErrorCode() {
                                 .errorCode(OutcomeXpayGatewayDto.ErrorCodeEnum.NUMBER_1)
                 )
                 .timestampOperation(OffsetDateTime.now());
-        AuthRequestDataUtils.AuthRequestData data = authRequestDataUtils.from(updateAuthorizationRequest);
+        TransactionId transactionId = new TransactionId(UUID.randomUUID().toString().replace("-", ""));
+        AuthRequestDataUtils.AuthRequestData data = authRequestDataUtils
+                .from(updateAuthorizationRequest, transactionId);
 
+        assertEquals(data.rrn(), ((OutcomeVposGatewayDto) updateAuthorizationRequest.getOutcomeGateway()).getRrn());
```
Comment (pietro-tota) [inline src/test/java/it/pagopa/transactions/utils/AuthRequestDataUtilsTest.java:117]:
```suggestion
```

**Review by pietro-tota — APPROVED** [review]
Just some comments

**Review by Fernando-Granato — COMMENTED** [review]
Need to change the tests the code review pipeline fails

**Review by pietro-tota — APPROVED** [review]

**Pull request merged**